### PR TITLE
Automatic update of version and changelog upon PR creation

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #77 

This PR solves the issue by logging in a github user and commit the changes to the version and changelog directly if it has not been done already.

As discussed in the issue, this is a feature that will (only) benefit simple PRs on develop. In the other cases (releases and refactor branches with fast forwarding) we need to manually take care of the version and changelog.

## Review guidelines

Estimated Time of Review: 5 minutes

### CODE REVIEW

**Comment by domire8** [comment]
As we can see, there are two new commits from the bot :+1:

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
           if ! [ "${VER_DIFF}" ]; then
             bash update_version.sh --commit && git push
           fi
-          CL_DIFF=$(git diff ${{ github.base_ref }} ${{ github.sha }} -- ./CHANGELOG.md)
+          CL_DIFF=$(git diff ${{ github.event.pull_request.base.sha }} ${{ github.sha }} -- ./CHANGELOG.md)
```
Comment (eeberhard) [inline .github/workflows/build-test.yml:33]: I saw you were still making changes so I held off my comment, but I was going to say switching to `.base_ref` seemed unnecessary when `event.pull_request.base.sha` was working fine before ;)

**Review by eeberhard — APPROVED** [review]
Ok, very cool. I think this is a helpful start. As you point out in the issue, I still think it will be more useful to set the version after closing the PR and before merging into the target branch, because A) the diff to review will be cleaner and B) we will always have synchronized versions. But, that is a separate issue for another time, and we can use this until then. (plus, re-use the logic from this change in that future implementation).

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (domire8) [reply .github/workflows/build-test.yml:33]: Hehehe

**Pull request merged**